File: api/src/main/java/com/delivery/api/service/TokenService.java
package com.delivery.api.service;

import com.delivery.common.dto.Token;
import com.delivery.common.exception.ApiException;
import com.delivery.common.helper.JwtTokenHelper;
import com.delivery.common.status.ErrorStatusCode;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Objects;

@RequiredArgsConstructor
@Service
public class TokenService {
    private final JwtTokenHelper jwtTokenHelper;

    public Token issueAccessToken(Long id) {
        var data = new HashMap<String, Object>();
        data.put("id", id);
        return jwtTokenHelper.issueAccessToken(data);
    }

    public Token issueRefreshToken(Long id) {
        var data = new HashMap<String, Object>();
        data.put("id", id);
        return jwtTokenHelper.issueRefreshToken(data);
    }

    public Long getIdOrThrow(String token) {
        var map = jwtTokenHelper.validate(token);
        var id = map.get("id");

        Objects.requireNonNull(id, () -> {
            throw new ApiException(ErrorStatusCode.NULL_POINT);
        });

        return Long.parseLong(id.toString());
    }
}

File: account/src/main/kotlin/com/delivery/account/service/TokenService.kt
package com.delivery.account.service

import com.delivery.account.component.JwtTokenHelper
import com.delivery.account.model.Token
import com.delivery.common.exception.ApiException
import com.delivery.common.status.ErrorStatusCode
import org.springframework.stereotype.Service

@Service
class TokenService(
    private val jwtTokenHelper: JwtTokenHelper
) {
    companion object {
        const val AUTHORIZATION_HEADER_PREFIX = "Bearer "
    }

    fun issueAccessToken(id: Long?): Token? {
        return id?.let {
            val data = mapOf("id" to id)
            jwtTokenHelper.issueAccessToken(data)
        }
    }

    fun issueRefreshToken(id: Long?): Token? {
        requireNotNull(id)

        return id.let {
            val data = mapOf("id" to id)
            jwtTokenHelper.issueRefreshToken(data)
        }
    }

    fun getIdOrThrow(token: String?): Long? {
        return token?.let { it ->
            jwtTokenHelper.validate(
                if (it.startsWith(AUTHORIZATION_HEADER_PREFIX)) {
                    it.replace(AUTHORIZATION_HEADER_PREFIX, "")
                } else {
                    it
                }
            )
        }?.let { map ->
            map["id"]
        }?.toString()?.toLong()
            ?: throw ApiException(ErrorStatusCode.NULL_POINT)
    }
}